python/usb_console/radio_session.py

```diff
@@ -59,27 +72,78 @@ def check_for_downlink(self):
         Read device output for debug messages and state variable updates. Record debug messages
         to the logging file, and update the console's record of the state.
         '''
+        
+        #connect to PAN email account  
+        mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+        mail.login(self.username, self.password)
+        mail.select('Inbox')
 
         while self.running_logger:
-            # TODO implement
-            pass
+            #.search() searches from mail. Data gives id's of all emails.
+            type, data = mail.search(None, '(FROM "sbdservice@sbd.iridium.com")', '(UNSEEN)')
+            mail_ids = data[0]
+            id_list = mail_ids.split()
+
+            for num in id_list:
```

Comment:
Can you also filter messages by IMEI number? The IMEI number is the subject line of the email.

The reason for this is that the PAN email address will receive data from both the follower and leader spacecraft, and a radio session should be able to distinguish between them.

Related: could you tell me how this code marks an email as unread?